extensions/WSLExtension/DistributionDefinitions/DistributionDefinitionHelper.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Runtime.InteropServices;
+using System.Text.Json;
+using Serilog;
+using Windows.Storage;
+using WSLExtension.ClassExtensions;
+using WSLExtension.Contracts;
+using WSLExtension.Helpers;
+using YamlDotNet.Serialization;
+using static WSLExtension.Constants;
+
+namespace WSLExtension.DistributionDefinitions;
+
+/// <summary>
+/// Provides definition information about all the WSL distributions that can be found at
+/// <see cref="Constants.KnownDistributionsWebJsonLocation"/>.
+/// </summary>
+public class DistributionDefinitionHelper : IDistributionDefinitionHelper
+{
+    private readonly ILogger _log = Log.ForContext("SourceContext", nameof(DistributionDefinitionHelper));
+
+    private readonly IHttpClientFactory _httpClientFactory;
+
+    private readonly PackageHelper _packageHelper = new();
+
+    private readonly Architecture _osArchitecture;
+
+    private readonly JsonSerializerOptions _jsonOptions = new()
+    {
+        PropertyNameCaseInsensitive = true,
+        PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+    };
+
+    public DistributionDefinitionHelper(IHttpClientFactory httpClientFactory)
+    {
+        _httpClientFactory = httpClientFactory;
+        _osArchitecture = RuntimeInformation.OSArchitecture;
+    }
+
+    /// <inheritdoc cref="IDistributionDefinitionHelper.GetDistributionDefinitionsAsync"/>
+    public async Task<Dictionary<string, DistributionDefinition>> GetDistributionDefinitionsAsync()
+    {
+        try
+        {
+            // Get the update to date distribution definitions from WSL GitHub repository.
+            var client = _httpClientFactory.CreateClient();
+            var distributionDefinitionsJson = await client.GetStringAsync(KnownDistributionsWebJsonLocation);
+            var webDefinitions = JsonSerializer.Deserialize<DistributionDefinitions>(distributionDefinitionsJson, _jsonOptions);
+            var distributionDefinitionsMap = new Dictionary<string, DistributionDefinition>();
+
+            foreach (var definition in webDefinitions!.Values)

```

Comment:
NIT: If you want to be fancy.

distributionDefinitionsMap.Add(webDefinitions!.Values.Where(x => x.ShouldAddDistributionx).ToDictionary()